cl/persistence/beacon_indexes/sql_indexer.go

```diff
@@ -0,0 +1,138 @@
+package beacon_indexes
+
+import (
+	"database/sql"
+	"fmt"
+
+	libcommon "github.com/ledgerwatch/erigon-lib/common"
+	"github.com/ledgerwatch/erigon/cl/cltypes"
+	_ "github.com/mattn/go-sqlite3"
+)
+
+// Define all queries templates
+var beaconIndiciesCreateTableQuery = `CREATE TABLE beacon_indicies (
+    Slot INTEGER NOT NULL,
+	BeaconBlockRoot BLOB NOT NULL CHECK(length(BeaconBlockRoot) = 32), -- Ensure it's 32 bytes
+	StateRoot BLOB NOT NULL CHECK(length(BeaconBlockRoot) = 32),
+	ParentBlockRoot BLOB NOT NULL CHECK(length(BeaconBlockRoot) = 32),
+    Canonical INTEGER NOT NULL DEFAULT 0, -- 0 for false, 1 for true
+    PRIMARY KEY (Slot, BeaconBlockRoot)  -- Composite key ensuring unique combination of Slot and BeaconBlockRoot
+);`
+
+var beaconIndiciesUniqueIndiciesQuery = `
+	CREATE UNIQUE INDEX idx_unique_canonical 
+	ON beacon_indicies (Slot) 
+	WHERE Canonical = 1;`
+
+var beaconReadBlockSlotByBlockRoot = "SELECT Slot FROM beacon_indicies WHERE BeaconBlockRoot = ?"
+
+var beaconReadParentRootByBlockRoot = "SELECT ParentBlockRoot FROM beacon_indicies WHERE BeaconBlockRoot = ?"
+
+var beaconReadCanonicalBlockRootBySlot = `SELECT BeaconBlockRoot FROM beacon_indicies 
+	WHERE Slot = ? AND Canonical = 1`
+
+var markRootCanonicalQuery = `UPDATE beacon_indicies 
+	SET Canonical = 1 
+	WHERE Slot = ? AND BeaconBlockRoot = ?`
+
+var resetOtherRootsCanonicalQuery = `UPDATE beacon_indicies 
+	SET Canonical = 0 
+	WHERE Slot = ? AND BeaconBlockRoot != ?`
+
+var insertOrUpdateBlockRoot = `
+	INSERT INTO beacon_indicies (Slot, BeaconBlockRoot, StateRoot, ParentBlockRoot, Canonical) 
+	VALUES (?, ?, ?, ?, 0);
+`
```

Comment:
this is an antipattern. put the raw query strings in the functions that they are used.


there are many reasons, but basically. 

1. this is just extra work for writer (you never reuse)
2. extra work for reader (another variable they need to jump to)
3. separates the logic from the code - makes your code more obscure/harder to read.

Reply:
notes on migrations - use "CREATE TABLE IF NOT EXISTS" and "CREATE UNIQUE INDEX IF NOT EXISTS"

later, we will create proper up and down migrations